python/test/executor_test.py

```diff
@@ -92,24 +93,40 @@ def main(
     npkit_dump_dir = os.getenv("NPKIT_DUMP_DIR")
     if npkit_dump_dir is not None:
         npkit.init(mscclpp_group.my_rank)
-    execution_plan = ExecutionPlan(execution_paln_name, execution_plan_path)
-
-    cp.random.seed(seed)
-    nelems = size // cp.dtype(dtype).itemsize
-    buffer = cp.random.random(nelems * mscclpp_group.nranks).astype(dtype)
-    sub_arrays = cp.split(buffer, MPI.COMM_WORLD.size)
-    sendbuf = sub_arrays[MPI.COMM_WORLD.rank]
-    expected = cp.zeros_like(sendbuf)
-    for i in range(mscclpp_group.nranks):
-        expected += sub_arrays[i]
+    execution_plan = ExecutionPlan(execution_plan_name, execution_plan_path)
+
+    if "allgather" in execution_plan_name:
+        cp.random.seed(seed)
+        nelems = size // cp.dtype(dtype).itemsize
+        buffer = cp.empty(nelems * mscclpp_group.nranks, dtype=dtype)
+        buffer[:] = cp.random.random(nelems * mscclpp_group.nranks, dtype=cp.float32).astype(dtype)
```

Comment:
Any reasons for running `empty()` and then `random()`?

Reply:
Ohhh, let me adjust this.